src/Servers/Kestrel/Core/src/Internal/Http2/HPack/IntegerEncoder.cs

```diff
@@ -46,7 +46,7 @@ public static bool Encode(int i, int n, Span<byte> buffer, out int length)
                         return false;
                     }
 
-                    i = i / 128;
+                    i /= 128;
```

Comment:
```suggestion
                    i >>= 7;    // i / 128
```

Or maybe cast to `uint` (the `while`-condition checks that no negative values are present), so the `%` above would also be optimized to the bit-hack by the JIT.

Otherwise apply `(byte)(i & 127 + 128)` by yourself?

Reply:
True, prefer the `uint` cast though.
```suggestion
                    i = ((uint)i) / 128;
```

Reply:
The cast should be near the top of the loop, so the modulo benefits from this too (L42).

Reply:
Changed